src/iio_wrapper.cpp

```diff
@@ -6,76 +6,81 @@
 #include <cstdlib>
 #include <sstream>
 
+struct iio_context* IIOWrapper::m_network_context = nullptr;
+
 IIOWrapper::IIOWrapper()
 {
+    if(m_network_context == nullptr)
+    {
+        m_network_context = iio_create_network_context("127.0.0.1");
+        m_object_context =  iio_context_clone(m_network_context);
+    }
+    else
+    {
+        m_object_context =  iio_context_clone(m_network_context);
+    }
+    m_dev = iio_context_find_device(m_object_context, "adis16505");
+    m_channel_accel_x = iio_device_find_channel(m_dev, "accel_x", false);
+    m_channel_accel_y = iio_device_find_channel(m_dev, "accel_y", false);
+    m_channel_accel_z = iio_device_find_channel(m_dev, "accel_z", false);
 
-}
-
-IIOWrapper::~IIOWrapper()
-{
+    m_channel_anglvel_x = iio_device_find_channel(m_dev, "anglvel_x", false);
+    m_channel_anglvel_y = iio_device_find_channel(m_dev, "anglvel_y", false);
+    m_channel_anglvel_z = iio_device_find_channel(m_dev, "anglvel_z", false);
 
 }
 
-std::string IIOWrapper::ssystem(const char *command,const char *filename)
+IIOWrapper::~IIOWrapper()
 {
+    iio_context_destroy(m_object_context);
```

Comment:
you also need to destroy the m_network_context. And also to make sure other instances of IIOWrapper will not attempt to destroy it as well.